test/e2e/commands/ingest_xmlupload/test_ingest_xmlupload.py

```diff
@@ -59,3 +62,24 @@ def _test_ingest_step(caplog: pytest.LogCaptureFixture, mapping_file: Path) -> N
 
     df = pd.read_csv(mapping_file)
     assert df["original"].tolist() == unordered(["testdata/bitstreams/test.jpg", "testdata/bitstreams/test.pdf"])
+
+
+def _test_xmlupload_step(caplog: pytest.LogCaptureFixture) -> None:
+    success = ingest_xmlupload(XML_FILE, CREDS)
+    assert success
+    logs = [rec.message for rec in caplog.records]
+    expected_logs = [
+        "The file 'mapping-4125.csv' is used to map the internal original filepaths to the internal image IDs.",
+        (
+            "All multimedia files referenced in the XML file were uploaded through dsp-ingest.\n"
+            "All multimedia files uploaded through dsp-ingest were referenced in the XML file."
+        ),
+    ]
+    assert all(expected_log in logs for expected_log in expected_logs)
+    expected_log_regexes = [
+        "Created resource 1/3",
+        "Created resource 2/3",
+        "Created resource 3/3",
+    ]
```

Comment:
I'm not a fan of testing against logs.  
If you feel it really helps, then I would suggest making it a very flexible test, something like `assert "mapping-4125.csv" in x`.  
But as it is currently, I fear this is more hindering than helpful.

Reply:
Good point. Now I don't test against logs any more. Rather, I test if the id2iri mapping file contains the correct resource IDs and IRIs: https://github.com/dasch-swiss/dsp-tools/commit/48d65acee20eb1b81dacfbe76f5637b9da0cfc4e